developer/src/server/build.sh

```diff
@@ -16,121 +10,68 @@ EX_USAGE=64
+  ":server     Keyman Developer Server main program" \
+  ":addins     Windows addins for GUI integration"
+
+builder_describe_internal_dependency \
+  publish:server       build:server \
+  publish:server       build:addins
+
+builder_describe_outputs \
+  configure:server     /node_modules \
+  configure:addins     /node_modules \
+  build:server         /developer/src/server/build/src/index.js \
+  build:addins         /developer/src/server/build/src/win32/console/index.js
+
+builder_parse "$@"
+
+#-------------------------------------------------------------------------------------------------------------------
+
+function clean_server() {
+  rm -rf "$THIS_SCRIPT_PATH/build"
+  rm -rf "$THIS_SCRIPT_PATH/node_modules"
+  rm -rf "$THIS_SCRIPT_PATH/tsconfig.tsbuildinfo"
+  rm -rf "$THIS_SCRIPT_PATH/src/site/resource"
+
+  # No longer in use, cleanup from previous versions:
+  rm -rf "$THIS_SCRIPT_PATH/dist"
 }
 
-display_usage ( ) {
-  echo "Usage: $0 [--production|--watch] [--test|--tdd]"
-  echo "       $0 --help"
-  echo
-  echo "  --help                      displays this screen and exits"
-  echo "  --production, -p            builds production release in /developer/bin/server/"
-  echo "  --skip-package-install, -S  don't run npm install (not valid with --production)"
-  echo "  --test, -t                  runs unit tests after building"
-  #echo "  --watch, -w                 builds dev server in watch mode"
-  echo "  --tdd                       runs unit tests WITHOUT building"
-  echo "  --no-build-addins           don't build/copy Win32 addins"
-  echo "  --no-build-kmw              don't build KeymanWeb"
-  echo "  --no-copy-kmw               don't copy KeymanWeb"
-}
-
-################################ Main script ################################
-
-run_tests=0
-production=0
-build=1
-install_dependencies=1
-build_keymanweb=1
-copy_keymanweb=1
-build_addins=1
-
-# Process command-line arguments
-while [[ $# -gt 0 ]] ; do
-  key="$1"
-  case $key in
-    --help|-h|-?)
-      display_usage
-      exit
-      ;;
-    --production|-p)
-      production=1
-      ;;
-    --skip-package-install|-S)
-      install_dependencies=0
-      ;;
-    --no-build-kmw)
-      build_keymanweb=0
-      ;;
-    --no-build-addins)
-      build_addins=0
-      ;;
-    --no-copy-kmw)
-      copy_keymanweb=0
-      ;;
-    --test)
-      run_tests=1
-      ;;
-    --tdd)
-      run_tests=1
-      build=0
-      install_dependencies=0
-      ;;
-    *)
-      echo "$0: invalid option: $key"
-      display_usage
-      exit $EX_USAGE
-  esac
-  shift # past the processed argument
-done
-
-# ----------------------------------------
-# Install dependencies
-# ----------------------------------------
-
-# Check if Node.JS/npm is installed.
-type npm >/dev/null ||\
-    builder_die "Build environment setup error detected!  Please ensure Node.js is installed!"
-
-if (( install_dependencies )) ; then
+function configure_server() {
   verify_npm_setup
```

Comment:
Does this handle installing dependencies? (install_dependences)

Reply:
Yes, that's now done in configure, and verify_npm_setup does the necessary `npm install`.